ompi/communicator/comm_cid.c

```diff
@@ -1026,7 +1026,7 @@ static int ompi_comm_allreduce_group_broadcast (ompi_comm_request_t *request)
 {
     ompi_comm_allreduce_context_t *context = (ompi_comm_allreduce_context_t *) request->context;
     ompi_comm_cid_context_t *cid_context = context->cid_context;
-    ompi_request_t *subreq[2];
```

Comment:
This is not necessary as subreq is a temporary array and the requests inside are copied using the subreq_count. Moreover, from a technical perspective these are supposed to be set to MPI_REQUEST_NULL and not to 0. This comment apply everywhere where such array is used.

Reply:
Ok, removed the 2 commits with these comm_cid fixes.